lib/event_sourcery/event_processing/esp_process.rb

```diff
@@ -26,6 +26,7 @@ def start
       rescue Exception => e
         EventSourcery.logger.fatal("An unhandled exception occurred in #{processor_name}")
         EventSourcery.logger.fatal(e)
+        EventSourcery.config.on_event_processor_error.call(e, processor_name)
```

Comment:
Do you think we should check to see if `on_event_processor_error` is defined before calling it, so consumers can choose whether to configure it or not?

Reply:
@scottyp-env Should be fine given a no-op default is defined:
https://github.com/envato/event_sourcery/blob/5269bc754e4357b1c94810139dca572f98426275/lib/event_sourcery/config.rb#L36-L47